Description: Change shell command from node to nodejs. Warn on failure.
 If a lifecycle script fails, it might be because "node" command was needed.
 We don't automatically try to fix it, and instead warn the user that might
 have been the cause, and where to find more information.
Forwarded: not-needed
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2012-09-21

Index: npm/scripts/index-build.js
===================================================================
--- npm.orig/scripts/index-build.js
+++ npm/scripts/index-build.js
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/nodejs
 var fs = require("fs")
   , path = require("path")
   , cli = path.resolve(__dirname, "..", "doc", "cli")
Index: npm/scripts/clean-old.sh
===================================================================
--- npm.orig/scripts/clean-old.sh
+++ npm/scripts/clean-old.sh
@@ -26,7 +26,7 @@ if [ "x$npm_config_prefix" != "x" ]; the
 else
   node="$NODE"
   if [ "x$node" = "x" ]; then
-    node=`which node`
+    node=`which nodejs`
   fi
   if [ "x$node" = "x" ]; then
     echo "Can't find node to determine prefix. Aborting."
Index: npm/scripts/install.sh
===================================================================
--- npm.orig/scripts/install.sh
+++ npm/scripts/install.sh
@@ -42,7 +42,7 @@ fi
 export npm_config_loglevel
 
 # make sure that node exists
-node=`which node 2>&1`
+node=`which nodejs 2>&1`
 ret=$?
 if [ $ret -eq 0 ] && [ -x "$node" ]; then
   (exit 0)
Index: npm/Makefile
===================================================================
--- npm.orig/Makefile
+++ npm/Makefile
@@ -1,4 +1,5 @@
 SHELL = bash
+node = nodejs
 
 markdowns = $(shell find doc -name '*.md' | grep -v 'index') README.md
 
@@ -33,23 +34,23 @@ latest:
 	@echo "Installing latest published npm"
 	@echo "Use 'make install' or 'make link' to install the code"
 	@echo "in this folder that you're looking at right now."
-	node cli.js install -g -f npm
+	$(node) cli.js install -g -f npm
 
 install: all
-	node cli.js install -g -f
+	$(node) cli.js install -g -f
 
 # backwards compat
 dev: install
 
 link: uninstall
-	node cli.js link -f
+	$(node) cli.js link -f
 
 clean: doc-clean uninstall
 	rm npmrc
-	node cli.js cache clean
+	$(node) cli.js cache clean
 
 uninstall:
-	node cli.js rm npm -g -f
+	$(node) cli.js rm npm -g -f
 
 doc: $(mandocs) $(htmldocs)
 
@@ -89,17 +90,17 @@ html/api/%.html: doc/api/%.md html/doche
 	scripts/doc-build.sh $< $@
 
 doc/cli/index.md: $(markdowns) scripts/index-build.js scripts/doc-build.sh package.json
-	node scripts/index-build.js > $@
+	$(node) scripts/index-build.js > $@
 
 node_modules/.bin/ronn:
-	node cli.js install ronn
+	$(node) cli.js install ronn
 
 doc: man
 
 man: $(cli_docs) $(api_docs)
 
 test:
-	node cli.js test
+	$(node) cli.js test
 
 publish: link doc
 	@git push origin :v$(shell npm -v) || true
Index: npm/cli.js
===================================================================
--- npm.orig/cli.js
+++ npm/cli.js
@@ -1,2 +1,2 @@
-#!/usr/bin/env node
+#!/usr/bin/nodejs
 require("./bin/npm-cli.js")
Index: npm/bin/npm-cli.js
===================================================================
--- npm.orig/bin/npm-cli.js
+++ npm/bin/npm-cli.js
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/nodejs
 ;(function () { // wrapper in case we're in module_context mode
 
 // windows: running "npm blah" in this folder will invoke WSH, not node.
Index: npm/lib/utils/lifecycle.js
===================================================================
--- npm.orig/lib/utils/lifecycle.js
+++ npm/lib/utils/lifecycle.js
@@ -164,7 +164,10 @@ function runPackageLifecycle (pkg, env,
       er.stage = stage
       er.script = env.npm_lifecycle_script
       er.pkgname = pkg.name
-      return cb(er)
+      cb(er)
+      log.warn('', 'This failure might be due to the use of legacy binary "node"')
+      log.warn("For further explanations, please read\n/usr/share/doc/nodejs/README.Debian\n")
+      return
     } else if (er) {
       log.error(pkg._id+"."+stage, er)
       log.error(pkg._id+"."+stage, "continuing anyway")
